### PR TITLE
Test protocol build on CI, fail if diffs

### PR DESCRIPTION
**Pull request by cjb**
@maxtaco r?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ script:
       docker build -t kbclient .;
     fi
   - cd ../shared && npm i --global flow-bin@`tail -n1 .flowconfig` && flow
+  - cd ../protocol && npm i && gem install activesupport && make clean && make && git diff --quiet --exit-code
```
Comment (maxtaco) [inline .travis.yml]: Can this be done before `./test/run_tests.sh`?  If there's a bug in the new AVDLs or the AVDL compiler, then that build will fail.

**Comment by cjb** [comment]
@maxtaco Thanks, made the change to do the protocol build first.

I'm only doing the protocol build on Linux because the OSX build machine only has Node 0.10 and we're using --harmony features.  That sound okay?  I can't think of a reason why we'd need to build it on both platforms.

**Comment by maxtaco** [comment]
Fine with me though I don't need harmony in current iced.

**Comment by cjb** [comment]
k, merging in

**Pull request merged**